handler/src/main/java/io/netty/handler/ssl/OpenSslClientSessionCache.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.ssl;
+
+import io.netty.internal.tcnative.SSL;
+import io.netty.util.AsciiString;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * {@link OpenSslSessionCache} that is used by the client-side.
+ */
+final class OpenSslClientSessionCache extends OpenSslSessionCache {
+    // TODO: Should we support to have a List of OpenSslSessions for a Host/Port key and so be able to
+    // support sessions for different protocols / ciphers to the same remote peer ?
+    private final Map<HostPort, NativeSslSession> sessions = new HashMap<HostPort, NativeSslSession>();
```

Comment:
no... it will re-use the same connection as long as its not a "single use".

Reply:
Right, so maybe the todo isn't so necessary?

Reply:
actually sorry no ... it may not be able to re-use the same session if there are different protocols / ciphers configured. Thats why I kept the TODO.